pacsman/dicom_interface_test.py

```diff
@@ -0,0 +1,149 @@
+from datetime import date
+
+import pytest
+from pydicom import Dataset
+
+from .dicom_interface import DicomInterface, PRIVATE_ID
+from .exceptions import InvalidDicomError
+
+
+def dataset_factory(defaults):
+    '''
+    Create a factory function for building pydicom datasets.
+
+    If any values are functions, they will be called to determine the value of
+    that attribute.  When the function is called, it will be passed an integer
+    indicating the number of times the factory has been called, and the full
+    attribute dict.
+
+    If a value is `None`, then it is not set.
+
+    :param defaults: Dict whose keys are tag names and values are tag values or functions.
+    :return: A factory function that produces datasets.  Keyword arguments
+        override the defaults.
+    '''
+    counter = 0
+
+    def factory(**overrides):
+        nonlocal counter
+        ds = Dataset()
+        attributes = {**defaults, **overrides}
+        for key, value in attributes.items():
+            if callable(value):
+                value = value(counter, attributes)
+            if value is not None:
+                setattr(ds, key, value)
+        counter += 1
+        return ds
+    return factory
+
+
+@pytest.fixture
+def slice_factory():
```

Comment:
Is "slice" the right term here?

Reply:
Good question; probably not ... I was thinking we may have more fixtures in the future that produce other various types of dicom datasets for testing.  E.g., we could have one that produces CT slices, one that produces MR slices, etc.

Of course, the current fixture only includes a small subset of the tags that would be in a required slice.

It may even make sense to have separate factories for different tests, so that each test is only excercising small datasets with just the necessary fields ... in that case, perhaps I should rename this to `patient_search_factory`?

Reply:
It seems more like `patient_dataset_factory` since we're not actually performing any searches around it.